packages/viem/src/utils/interop.ts

```diff
@@ -0,0 +1,79 @@
+import type {
+  Account,
+  Chain,
+  Hash,
+  Log,
+  PublicClient,
+  TransactionReceipt,
+  Transport,
+} from 'viem'
+import { concat, parseEventLogs } from 'viem'
+
+import { crossL2InboxABI } from '@/abis.js'
+import type { MessageIdentifier, MessagePayload } from '@/types/interop.js'
+
+export type CreateInteropMessageParameters = { log: Log }
+export type CreateInteropMessageReturnType = {
+  id: MessageIdentifier
+  payload: MessagePayload
+}
+
+export type DecodeExecutingMessagesParameters = { receipt: TransactionReceipt }
+export type DecodeExecutingMessagesReturnType = {
+  executingMessages: Array<{ id: MessageIdentifier; msgHash: Hash; log: Log }>
+}
+
+/**
+ * Utility for creating an interoperable {@link MessageIdentifier} & {@link MessagePayload} from a log
+ * @category Utils
+ * @param params {@link CreateInteropMessageParameters}
+ * @returns created interop message {@link Message}
+ */
+export async function createInteropMessage<
+  chain extends Chain | undefined,
+  account extends Account | undefined,
+>(
+  client: PublicClient<Transport, chain, account>,
+  params: CreateInteropMessageParameters,
+): Promise<CreateInteropMessageReturnType> {
+  const { log } = params
+  if (log.blockNumber === undefined || log.logIndex === undefined) {
+    throw new Error('pending log cannot be constructed into an interop message')
+  }
+  if (!client.chain) {
+    throw new Error('define chain required to construct an interop message')
+  }
+
+  const block = await client.getBlock({ blockHash: log.blockHash as Hash })
+  const id: MessageIdentifier = {
+    origin: log.address,
+    logIndex: BigInt(log.logIndex!),
+    blockNumber: block.number,
+    timestamp: block.timestamp,
+    chainId: BigInt(client.chain.id),
+  }
+
+  const payload = concat([...params.log.topics, params.log.data])
```

Comment:
does the 0x have to be sliced off the front of each topic and log.data? something like:
```
function executingMessagePayloadBytes(log: {
  topics: string[]
  data: string
}): `0x${string}` {
  let msg = ''

  // Concatenate each topic (removing the "0x" prefix from each)
  for (const topic of log.topics) {
    msg += topic.startsWith('0x') ? topic.slice(2) : topic
  }

  // Concatenate log.data (removing the "0x" prefix if it exists)
  const logData = log.data.startsWith('0x') ? log.data.slice(2) : log.data

  // Return the concatenated result prefixed with "0x"
  return ('0x' + msg + logData) as `0x${string}`
}
```

Reply:
Yep it does, usage:   https://viem.sh/docs/utilities/concat.html#usage

Reply:
that's awesome, didnt know this existed